packages/foundry/src/vault/adapter/aave/aaveV2/AaveV2Adapter.sol

```diff
@@ -0,0 +1,162 @@
+import { ILendingPool, IAaveMining, IAToken } from "./IAaveV2.sol";
+
+/**
+ * @title   AaveV2 Adapter
+ * @author  amatureApe
+ * @notice  ERC4626 wrapper for AaveV2 Vaults.
+ *
+ * An ERC4626 compliant Wrapper for https://github.com/aave/protocol-v2/blob/master/contracts/protocol/lendingpool/LendingPool.sol.
+ * Allows wrapping AaveV2 aTokens with or without an active Liquidity Mining.
+ * Allows for additional strategies to use rewardsToken in case of an active Liquidity Mining.
+ */
+
+contract AaveV2Adapter is AdapterBase, WithRewards {
+  using SafeERC20 for IERC20;
+  using Math for uint256;
+
+  string internal _name;
+  string internal _symbol;
+
+  // @notice The Aave aToken contract
+  IAToken public aToken;
+
+  // @notice The Aave liquidity mining contract
+  IAaveMining public aaveMining;
+
+  // @notice Check to see if Aave liquidity mining is active
+  bool public isActiveMining;
+
+  // @notice The address that will receive the liquidity mining rewards if LM is active
+  address public rewardsRecipient;
+
+  // @notice The Aave LendingPool contract
+  ILendingPool public lendingPool;
+
+  /*//////////////////////////////////////////////////////////////
+                            CONSTRUCTOR
+    //////////////////////////////////////////////////////////////*/
+
+  /**
+   * @notice Initialize a new AaveV2 Adapter.
+   * @param adapterInitData Encoded data for the base adapter initialization.
+   * @param aaveInitData Encoded data for the beefy adapter initialization.
+   * @dev `_aToken` - The underlying asset supplied to and wrapped by Aave.
+   * @dev `_lendingPool` - The lending pool.
+   * @dev `_aaveMining` - An optional liquidity mining contract to boost yield.
+   * @dev This function is called by the factory contract when deploying a new vault.
+   */
+  function initialize(
+    bytes memory adapterInitData,
+    address,
+    bytes memory aaveInitData
+  ) public {
+    (address _aToken, address _lendingPool, address _aaveMining, address _rewardsRecipient) = abi.decode(
+      aaveInitData,
+      (address, address, address, address)
+    );
+    __AdapterBase_init(adapterInitData);
+
+    _name = string.concat("Popcorn AaveV2", IERC20Metadata(asset()).name(), " Adapter");
+    _symbol = string.concat("popB-", IERC20Metadata(asset()).symbol());
+
+    aToken = IAToken(_aToken);
+    lendingPool = ILendingPool(_lendingPool);
+    aaveMining = IAaveMining(_aaveMining);
+    // Method implemented manually inputs mining contract and then checks for emission.
+    // Commented method below queries the aToken for contract.
+    // **IAaveIncentivesController aaveMining = aToken.getIncentivesController();**
+    rewardsRecipient = _rewardsRecipient;
+
+    IERC20(asset()).approve(_lendingPool, type(uint256).max);
+
+    (, uint256 emission, ) = aaveMining.getAssetData(asset());
+
+    isActiveMining = emission > 0 ? true : false;
+  }
+
+  function name() public view override(IERC20Metadata, ERC20) returns (string memory) {
```

Comment:
Do we need to override these? I think the AdapterBase does this already.

Reply:
I was using the BeefyAdapter as a guide which had them included, but you're right IERC20MetadataUpgradeable in the AdapterBase should already take care of it